samples/localities-search/index.ts

```diff
@@ -0,0 +1,279 @@
+// [START woosmap_localities_search]
+let map: woosmap.map.Map;
+let marker: woosmap.map.Marker;
+let infoWindow: woosmap.map.InfoWindow;
+let localitiesService: woosmap.map.LocalitiesService;
+let debouncedLocalitiesSearch: (...args: any[]) => Promise<any>;
+let input: string;
+let detailsHTML: HTMLElement;
+let addressDetailsContainer: HTMLElement;
+
+const woosmap_key = "YOUR_API_KEY";
+
+function initMap(): void {
+  map = new window.woosmap.map.Map(
+    document.getElementById("map") as HTMLElement,
+    {
+      center: { lat: 51.507445, lng: -0.127765 },
+      zoom: 8,
+    }
+  );
+  infoWindow = new woosmap.map.InfoWindow({});
+  localitiesService = new window.woosmap.map.LocalitiesService();
+
+  debouncedLocalitiesSearch = debouncePromise(fetchLocalitiesSearch, 0);
+}
+
+const fetchLocalitiesSearch = async (input: any): Promise<any> => {
+  const center = map.getCenter();
+  const radius = (map.getZoom() > 10) ? "10000" : "100000"
+  try {
+    const response = await fetch(
+      `
+https://api.woosmap.com/localities/search?types=point_of_interest|locality|admin_level|postal_code|address&input=${encodeURIComponent(input)}&location=${center.lat()},${center.lng()}&radius=${radius}&key=${woosmap_key}&components=country%3Agb`
+    );
+    return await response.json();
+  } catch (error) {
+    console.error("Error fetching localities:", error);
+    throw error;
+  }
+};
+
+function fillAddressDetails(
+  addressDetails: any
+) {
+  const details: string[] = [];
+  detailsHTML.innerHTML = "";
+  detailsHTML.style.display = "block";
+  if (addressDetails.formatted_address) {
+    details.push(
+      `<p class='option-detail'><span class='option-detail-label'>Formatted_address:</span><span class='bold'>${addressDetails.formatted_address}</span></p>`
+    );
+  } else if (addressDetails.name)
+    details.push(
+      `<p class='option-detail'><span class='option-detail-label'>Name:</span><span class='bold'>${addressDetails.name}</span></p>`
+    );
+  if (addressDetails.types && addressDetails.types[0]) {
+    details.push(
+      `<p class='option-detail'><span class='option-detail-label'>Type: </span><span class='bold'>${addressDetails.types[0]}</span></p>`
+    );
+  }
+  if (addressDetails.categories)
+    details.push(
+      `<p class='option-detail'><span class='option-detail-label'>Categories:</span><span class='bold'>${addressDetails.categories[0]}</span></p>`
+    );
+  if (addressDetails.geometry) {
+    details.push(
+      `<div class='option-detail'><div><span class='option-detail-label'>Latitude:</span> <span class='bold'>${addressDetails.geometry.location.lat.toString()}</span></div><div><span class='option-detail-label'>Longitude: </span><span class='bold'>${addressDetails.geometry.location.lng.toString()}</span></div></div>`
```

Comment:
We could round the lat/lng to 5 decimals just to make it nicer on the details display